File: app/src/main/java/com/saizad/mvvmexample/components/auth/AuthFragment.java
package com.saizad.mvvmexample.components.auth;


import com.saizad.mvvmexample.api.AuthApi;
import com.saizad.mvvmexample.components.MVVMExampleFragment;
import com.saizad.mvvmexample.di.auth.AuthEnvironment;

import javax.inject.Inject;


public abstract class AuthFragment<V extends AuthViewModel> extends MVVMExampleFragment<V> {

    @Inject
    public AuthEnvironment authEnvironment;

    protected AuthApi api() {
        return authEnvironment.api();
    }

}

File: app/src/main/java/com/vm/frameworkexample/components/auth/AuthFragment.kt
package com.vm.frameworkexample.components.auth

import com.vm.framework.VmFrameworkNetworkRequest
import com.vm.frameworkexample.components.VmFrameworkExampleFragment
import com.vm.frameworkexample.di.auth.AuthEnvironment
import javax.inject.Inject

abstract class AuthFragment<V : AuthViewModel> : VmFrameworkExampleFragment<V>() {
    @Inject
    lateinit var authEnvironment: AuthEnvironment

    override val networkRequest: VmFrameworkNetworkRequest by lazy {
        authEnvironment.networkRequest
    }
}

